ui/lib/src/elements/where-space-dialog.ts

```diff
@@ -0,0 +1,101 @@
+import { html, css, LitElement } from "lit";
+import { property } from "lit/decorators.js";
+
+import { sharedStyles } from "../sharedStyles";
+import { contextProvided } from "@lit-labs/context";
+import { ScopedElementsMixin } from "@open-wc/scoped-elements";
+import { WhereStore } from "../where.store";
+import { whereContext, Space } from "../types";
+import { Dialog, TextField, Button, Checkbox, Formfield } from "@scoped-elements/material-web";
+
+/**
+ * @element where-space
+ */
+export class WhereSpaceDialog extends ScopedElementsMixin(LitElement) {
+  @property({ type: String }) current = "";
+
+  /** Dependencies */
+  @contextProvided({ context: whereContext })
+  _store!: WhereStore;
+
+  open() {
+    const dialog = this.shadowRoot!.getElementById("space-dialog") as Dialog
+    dialog.open = true
+  }
+
+  private async handleSpaceDialog(e: any) {
+    const name = this.shadowRoot!.getElementById(
+      "space-dialog-name"
+    ) as TextField;
+    const url = this.shadowRoot!.getElementById(
+      "space-dialog-url"
+    ) as TextField;
+    const multi = this.shadowRoot!.getElementById(
+      "space-dialog-multi"
+    ) as Checkbox;
+    if (e.detail.action == "ok") {
+      const img = this.shadowRoot!.getElementById("sfc") as HTMLImageElement;
+      img.onload = async () => {
+        const space: Space = {
+          name: name.value,
+          surface: {
+            url: url.value,
+            size: { x: img.naturalHeight, y: img.naturalWidth },
+            data: "[]",
+          },
+          meta: {
+            multi: multi.checked ? "true" : "",
+          },
+          wheres: [],
+        };
+        this.current = await this._store.addSpace(space);
+        name.value = "";
+        url.value = "";
+      };
+      img.src = url.value;
+    } else {
+      name.value = "";
+      url.value = "";
+    }
+  }
+
+  handleUrlUpdated(e:Event) {
+    console.log(e)
+  }
+
+  render() {
+    return html`
+<mwc-dialog  id="space-dialog" heading="Space" @closing=${
+this.handleSpaceDialog
+}>
+<mwc-textfield id="space-dialog-name" minlength="3" maxlength="64" placeholder="Name" required></mwc-textfield>
+<mwc-textfield @input={$this.handleUrlUpdated} id="space-dialog-url" placeholder="Image URL" required></mwc-textfield>
```

Comment:
Careful with the syntax (this would be catched at first glance by typescript plugin in and IDE...). 

This is like `@input=${this.handleUrlUpdated}`, and then in the function it's something like `e.target.value`